x/rewards/keeper/iprpc_test.go

```diff
@@ -0,0 +1,642 @@
+
+	// make a provider service an IPRPC eligible consumer and advance a month
+	// there should be no iprpc rewards objects
+	c1Acc, _ := ts.GetAccount(common.CONSUMER, 0)
+	_, p1 := ts.GetAccount(common.PROVIDER, 0)
+	relay := ts.SendRelay(p1, c1Acc, []string{ts.specs[1].Index}, 100)
+	_, err = ts.Servers.PairingServer.RelayPayment(ts.GoCtx, &relay)
+	require.NoError(t, err)
+	ts.AdvanceMonths(1)
+	ts.AdvanceEpoch()
+	res, err = ts.QueryRewardsIprpcSpecReward(mockSpec2)
+	require.NoError(t, err)
+	require.Len(t, res.IprpcRewards, 0)
+	require.Equal(t, uint64(3), res.CurrentMonthId)
+}
+
+// TestIprpcMinCost tests that a fund TX fails if it doesn't have enough tokens to cover for the minimum IPRPC costs
+// Scenarios:
+// 1. fund TX with the minimum cost available -> TX success
+// 2. assume min cost = 100ulava, fund TX with 50ulava and 200ibc -> TX fails (ibc "has enough funds")
+// 3. fund TX without the minimum cost available -> TX fails
+// 4. fund TX with the minimum cost but creator doesn't have enough balance for the funding -> TX fails
+func TestIprpcMinCost(t *testing.T) {
+	ts := newTester(t, true)
+	ts.setupForIprpcTests(false)
+	consumerAcc, consumer := ts.GetAccount(common.CONSUMER, 0)
+	err := ts.Keepers.BankKeeper.AddToBalance(consumerAcc.Addr, sdk.NewCoins(sdk.NewCoin(ibcDenom, sdk.NewInt(500))))
+
+	_, poorConsumer := ts.AddAccount(common.CONSUMER, 1, minIprpcCost.Amount.Int64()-10)
+
+	testCases := []struct {
+		name    string
+		creator string
+		fund    sdk.Coins
+		success bool
+	}{
+		{
+			name:    "Happy flow - creator with enough funds and above min iprpc cost",
+			creator: consumer,
+			fund:    sdk.NewCoins(minIprpcCost.AddAmount(sdk.NewInt(10))),
+			success: true,
+		},
+		{
+			name:    "fund without min iprpc cost",
+			creator: consumer,
+			fund:    sdk.NewCoins(minIprpcCost.SubAmount(sdk.NewInt(10))),
+			success: false,
+		},
+		{
+			name:    "fund with other denom above min iprpc cost",
+			creator: consumer,
+			fund:    sdk.NewCoins(sdk.NewCoin(ibcDenom, minIprpcCost.Amount.AddRaw(10))),
+			success: false,
+		},
+		{
+			name:    "insufficient balance for fund",
+			creator: poorConsumer,
+			fund:    sdk.NewCoins(minIprpcCost.AddAmount(sdk.NewInt(10))),
+			success: false,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err = ts.TxRewardsFundIprpc(tt.creator, mockSpec2, 1, tt.fund)
+			if tt.success {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+			}
+		})
+	}
+}
+
+// TestIprpcEligibleSubscriptions tests that IPRPC CU is counted only if serviced an eligible subscription
+// Scenarios:
+// 0. assume two providers: p1, p2 and two consumers: c1, c2. Only c1 is IPRPC eligible
+// 1. p1 provides service for both consumers, p2 provides service for c1 -> IPRPC reward should divide equally between p1 and p2
+// 2. both providers provide service for c2 -> No IPRPC rewards should be given
```

Comment:
let's add another month after the funding ends and see they are not getting more rewards after

Reply:
done. see https://github.com/lavanet/lava/commit/85ba6891d7cb69e895c6500e8115b27858e3a3cb